Hash registry contract names

bricewang opened the pull request:
### Description

Replaces registry ids in `UsingRegistry` with their packed `keccak256` hashes. Changes `getAddressFor` and `getAddressForOrDie` in `Registry` to take a `bytes32` hash. Changes tests using these functions to hash the name before passing it in.

### Tested

Tests refactored, no new ones needed.

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/527
- Sister PR of https://github.com/celo-org/celo-blockchain/pull/418

bricewang committed:
Hash registry contract names

bricewang committed:
Repoint CircleCI to associated celo-blockchain changes

yerdua commented:
`contractkit` relies on being able to look up contract addresses from the registry, so this PR also necessitates changes to that code. `getAddressFor()` is called here: https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/address-registry.ts#L21-L33

And this is where it has a mapping of contract name to registry key: https://github.com/celo-org/celo-monorepo/blob/master/packages/contractkit/src/base.ts#L3-L19

m-chrzan committed:
Merge branch 'master' into bricewang/registry-lookup

m-chrzan committed:
Hash Registry IDs in contractkit

m-chrzan commented:
I'm taking over this branch. Brice's changes look good to me, fixed up the thing Audrey pointed out. Waiting for another approval.

timmoreton reviewed (approved):
What is the plan to merge with this and the other PR? Take it you’ll merge both then raise a new PR to revert the circleci changes?

m-chrzan committed:
Appease the linter

m-chrzan committed:
Merge branch 'master' into bricewang/registry-lookup

drklee3 committed:
[wallet] Return max length if decimal pos is 0 (#776)

Prevents decimal inputs after the set max length for numbers that begin with a decimal point. Removed an unnecessary truthy check which caused the problem.

Commit:
yarn build in celotool Dockerfile (#779)

asaj commented on packages/protocol/contracts/common/Registry.sol:
```diff
@@ -38,25 +38,19 @@ contract Registry is IRegistry, Ownable, Initializable {
   /**
    * @notice Gets address associated with the given identifier.
-   * @param identifier Identifier of contract whose address we want to look up.
+   * @param identifier Identifier hash of contract whose address we want to look up.
```
Since these are view calls should we continue to allow users to submit the string?

mcortesi replied:
I agree. more precisely, i think we can keep the old interface for everything. And thus have "retrocompatibility" :D

m-chrzan replied:
While in this case having the string functions might be useful, I don't know that the reason for keeping them should be backwards compatibility. We don't need to remain backwards compatible with Alfajores, right? If anything, if we do find APIs that could be improved, we should completely replace them so that by mainnet we have as clean an interface as possible.

mcortesi replied:
Sure, but what i propose is to do something like:

1. Make change, leave old api, mark it as deprectaded
2. Deploy
3. On a later PR, after all apps have already migrated to new contractkit/etc code that call new methods, remove deprecated methods

jeanregisser committed:
[wallet] Log calculate fee errors (#799)

### Description

Investigating #245 showed calculate fee errors trigger the alert banner with the generic "Could not calculate fee" but don't log the actual underlying error. This fixes it.

### Tested

Calculate Fee errors are now logged.

### Other changes

N/A

### Related issues

- #245

### Backwards compatibility

Yes.

jmrossy committed:
[Wallet] Faster fees and other fee related UX fixes (#777)

* Add fee and new balance line items to SendAmount screen
* Add caching of fees
* Fix fee estimation for sending invites
* Break out invite fees into gas and verification fee on review screen
* Trigger fee estimation on Send screen to speed up SendAmount
* De-sagafy get gas price utility to facilitate access from hooks
* Split the TransferConfirmationCard component into two
* Add Fee estimation saga tests
* Add a reusable LoadingLabel component which shows a spinner until a value is ready
* Some cleanup of the utils/formatting functions

jeanregisser committed:
[wallet] upgrade react-redux (#812)

* Upgrade react-redux to latest stable version

This is so we can use the new hooks API
Also removed duplicated redux / react-redux versions

* Upgrade failing tests with the react-redux upgrade

- Use bindActionsCreator instead of the shorthand syntax for some
mapDispatchToProps that were using redux-thunk actions to workaround a
new typescript error with the upgraded react-redux definitions
- Migrated some tests to react-native-testing-library as they
were raising 'TypeError: ShallowWrapper::dive() can only be called on components'.
This is because the enzyme version we're using is not compatible with the new react
context API used by the new react-redux

* Ignore a new typescript error

* Simplify CalculateFee by using useDispatch instead of passing showError down to the component

m-chrzan committed:
Expose registry entries by contract name

m-chrzan committed:
Merge branch 'master' into bricewang/registry-lookup

mcortesi commented on .circleci/config.yml:
```diff
@@ -410,7 +410,7 @@ jobs:
             go version
             cd packages/celotool
             mkdir ~/.ssh/ && echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
-            ./ci_test_sync_with_network.sh checkout master
```
do we need this?

m-chrzan replied:
This is temporary to get tests passing on this PR, making it mergeable. The idea would be to immediately after this and the geth PR get merged, to repoint back to master.

asaj replied:
+1 to @mcortesi, if we merge the celo-blockchain PR first we can keep this pointed to master (or vice versa)

mcortesi commented on packages/protocol/test/common/integration.ts:
```diff
@@ -11,6 +11,7 @@ import {
   ReserveInstance,
   StableTokenInstance,
 } from 'types'
+const soliditySha3 = new (require('web3'))().utils.soliditySha3
```
can't we use `Web3.utils.soliditySha3` and also avoid the `require`?

asaj replied:
+1, we should be able to use the `web3` provided to us by truffle

asaj commented on packages/protocol/contracts/common/Registry.sol:
```diff
@@ -29,34 +29,46 @@ contract Registry is IRegistry, Ownable, Initializable {
-    );
-    require(registry[hash] != address(0), "identifier has no registry entry");
-    return registry[hash];
+  function getAddressForOrDieString(string calldata identifier) external view returns (address) {
```
nit: consider `getAddressForStringOrDie` instead

m-chrzan committed:
Merge branch 'master' into bricewang/registry-lookup

m-chrzan committed:
Don't reimport soliditySha3

m-chrzan committed:
Rename function

m-chrzan committed:
Merge branch 'master' into bricewang/registry-lookup

m-chrzan committed:
Repoint CI to master
